pyiceberg/types.py

```diff
@@ -62,6 +63,12 @@
 FIXED_PARSER = ParseNumberFromBrackets(FIXED)
 
 
+class TableVersion(IntEnum):
+    ONE = 1
+    TWO = 2
+    THREE = 3
```

Comment:
interesting to see this, we were talking about something like this in #851

Reply:
I don't see a ton of added value in addition to using a `TypeAlias` like we have in typedef:

https://github.com/apache/iceberg-python/blob/c06e32034fbd32d5aab9b2ef19ad2d253141479d/pyiceberg/typedef.py#L225

But I don't want to be that guy blocking this

Reply:
hm yea, now we have 2 `TableVersion`. one in `types.py`, the other in [`typedef.py`](https://github.com/apache/iceberg-python/blob/c06e32034fbd32d5aab9b2ef19ad2d253141479d/pyiceberg/typedef.py#L225)

Reply:
this is only used in `minimum_format_version`. maybe we can use the `types.TableVersion` int literal for now and punt the enum discussion to #851

```
    def minimum_format_version(self) -> TableVersion:
        """Minimum Iceberg format version after which this type is supported."""
        return 1
```

Reply:
Yeah I didn't know the `TypeAlias` existed - I've removed the `IntEnum` and used the `TypeAlias` instead

Reply:
No problem, let's defer that to #851